Tests/HealthKitOnFHIRTests/HKCategorySampleTests.swift

```diff
@@ -0,0 +1,472 @@
+                    display: "Ovulation Test Result"
+                )
+            )
+            XCTAssertEqual(
+                observation.value, .string(value.description.asFHIRStringPrimitive())
+            )
+        }
+    }
+
+    func testContraceptive() throws {
+        let values: [HKCategoryValueContraceptive] = [.unspecified, .implant, .injection, .intrauterineDevice, .intravaginalRing, .oral, .patch]
+
+        for value in values {
+            let observation = try createObservationFrom(
+                type: HKCategoryType(.contraceptive),
+                value: value.rawValue
+            )
+
+            XCTAssertEqual(
+                observation.code.coding?.first,
+                createCategoryCoding(
+                    categoryType: HKCategoryType(.contraceptive).description,
+                    display: "Contraceptive"
+                )
+            )
+            XCTAssertEqual(observation.value, .string(value.description.asFHIRStringPrimitive()))
+        }
+    }
+
+    func testSleepAnalysis() throws {
+        let values: [HKCategoryValueSleepAnalysis] = [.inBed, .asleepUnspecified, .awake, .asleepCore, .asleepDeep, .asleepREM]
+
+        for value in values {
+            let observation = try createObservationFrom(
+                type: HKCategoryType(.sleepAnalysis),
+                value: value.rawValue
+            )
+
+            XCTAssertEqual(
+                observation.code.coding?.first,
+                createCategoryCoding(
+                    categoryType: HKCategoryType(.sleepAnalysis).description,
+                    display: "Sleep Analysis"
+                )
+            )
+            XCTAssertEqual(observation.value, .string(value.description.asFHIRStringPrimitive()))
+        }
+    }
+
+    func testAppetiteChanges() throws {
+        let values: [HKCategoryValueAppetiteChanges] = [.unspecified, .noChange, .decreased, .increased]
+
+        for value in values {
+            let observation = try createObservationFrom(
+                type: HKCategoryType(.appetiteChanges),
+                value: value.rawValue
+            )
+
+            XCTAssertEqual(
+                observation.code.coding?.first,
+                createCategoryCoding(
+                    categoryType: HKCategoryType(.appetiteChanges).description,
+                    display: "Appetite Changes"
+                )
+            )
+            XCTAssertEqual(observation.value, .string(value.description.asFHIRStringPrimitive()))
+        }
+    }
+
+    func testEnvironmentalAudioExposureEvent() throws {
+        let observation = try createObservationFrom(
+            type: HKCategoryType(.audioExposureEvent),
+            value: HKCategoryValueEnvironmentalAudioExposureEvent.momentaryLimit.rawValue
+        )
+
+        XCTAssertEqual(
+            observation.code.coding?.first,
+            createCategoryCoding(
+                categoryType: HKCategoryType(.audioExposureEvent).description,
```

Comment:
To remove the warning:
```suggestion
                categoryType: HKCategoryType(.environmentalAudioExposureEvent).description,
```

Reply:
See comment above.